File: livingdoc-fixture-converter/src/test/java/org/livingdoc/fixture/converter/common/StringConverterTest.java
package org.livingdoc.fixture.converter.common;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;


class StringConverterTest {

    StringConverter cut = new StringConverter();

    @ParameterizedTest(name = "\"{0}\"")
    @ValueSource(strings = { "", " ", "foo", "foo bar" })
    void anyStringValueIsReturnedAsIs(String value) {
        assertThat(cut.convert(value)).isEqualTo(value);
    }

}

File: livingdoc-fixture-converter/src/test/java/org/livingdoc/fixture/converter/common/StringConverterTest.kt
package org.livingdoc.fixture.converter.common

import org.assertj.core.api.Assertions.assertThat

import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource


internal class StringConverterTest {

    val cut = StringConverter()

    @ParameterizedTest(name = "\"{0}\"")
    @ValueSource(strings = arrayOf("", " ", "foo", "foo bar"))
    fun `any string value is returned as is`(value: String) {
        assertThat(cut.convert(value)).isEqualTo(value)
    }

}
